### PR TITLE
Check if auth is missing on quantum architecture fetch

### PR DESCRIPTION
**Pull request by kukushechkin**
Fetching the quantum architecture is the first step in e.g. `iqm-qiskit` and should provide a meaningful error message is the auth credentials are missing.

### CODE REVIEW

File: tests/test_authentication.py [hunk 1]
```diff
@@ -69,6 +88,52 @@ def test_add_authorization_header_when_credentials_are_provided(base_url, creden
     unstub()
 
 
+def test_add_authorization_header_on_submit_circuits_when_credentials_are_provided(
+    base_url, credentials, sample_circuit
+):
+    """
+    Tests that requests are sent with Authorization header when credentials are provided
```
Comment (freetonik) [inline tests/test_authentication.py:95]: Comments on lines 81, 95, 109, 123 are slightly confusing in their similarity :)

File: tests/test_authentication.py [hunk 2]
```diff
@@ -69,6 +88,52 @@ def test_add_authorization_header_when_credentials_are_provided(base_url, creden
...
+
+
+def test_submit_circuits_raises_when_no_credentials_are_provided(base_url, credentials, sample_circuit):
+    """
+    Tests that requests are sent with Authorization header when credentials are provided
+    """
+    tokens = prepare_tokens(300, 3600, **credentials)
+    expect_submit_circuits_request(base_url, tokens['access_token'], response_status=200)
+    client = IQMClient(base_url, **credentials)
+    client.submit_circuits(
```
Comment (freetonik) [inline tests/test_authentication.py:114]: Should there be `with...` and `assert`?

File: tests/test_authentication.py [hunk 3]
```diff
@@ -57,9 +65,20 @@ def test_get_initial_tokens_with_credentials_from_env_variables(base_url, creden
...
+    with pytest.raises(ClientConfigurationError) as e:
+        IQMClient(base_url)
+    assert str(e.value) == 'Auth server URL is set but no username or password'
+    unstub()
+
+
 def test_add_authorization_header_when_credentials_are_provided(base_url, credentials):
     """
-    Tests that requests are sent with Authorization header when credentials are provided
+    Tests that `get_run` requests are sent with Authorization header when credentials are provided
```
Comment (freetonik) [inline tests/test_authentication.py:81]:
```suggestion
    Tests that ``get_run`` requests are sent with Authorization header when credentials are provided
```

I can't remember the exact edge cases, but we use double backticks in Markdown... I think it's for better docs rendering (when inline code is placed inside a docstring or something)

File: tests/test_authentication.py [hunk 4]
```diff
@@ -69,6 +88,41 @@ def test_add_authorization_header_when_credentials_are_provided(base_url, creden
     unstub()
 
 
+def test_add_authorization_header_on_submit_circuits_when_credentials_are_provided(
+    base_url, credentials, sample_circuit
+):
+    """
+    Tests that `submit_circuits` requests are sent with Authorization header when credentials are provided
```
Comment (freetonik) [inline tests/test_authentication.py:95]:
```suggestion
    Tests that ``submit_circuits`` requests are sent with Authorization header when credentials are provided
```

Comment (freetonik) [inline tests/test_authentication.py:95]:
```suggestion
    Tests that ``submit_circuits`` requests are sent with Authorization header when credentials are provided
```

File: tests/test_authentication.py [hunk 5]
```diff
@@ -69,6 +88,41 @@ def test_add_authorization_header_when_credentials_are_provided(base_url, creden
...
+        qubit_mapping={'Qubit A': 'QB1', 'Qubit B': 'QB2'},
+        shots=1000,
+    )
+    assert expected_job_id == created_job_id
+    unstub()
+
+
+def test_submit_circuits_raises_when_auth_failed(base_url, credentials, sample_circuit):
+    """
+    Tests that submit_circuits raises when authentication fails
```
Comment (freetonik) [inline tests/test_authentication.py:111]:
```suggestion
    Tests that ``submit_circuits`` raises ClientAuthenticationError when authentication fails
```

**Review by freetonik — APPROVED** [review]

**Review by freetonik — APPROVED** [review]
